airflow/io/path.py

```diff
@@ -89,7 +82,7 @@ class ObjectStoragePath(CloudPath):
         "_hash",
     )
 
-    def __new__(cls: type[PT], *args: str | os.PathLike, **kwargs: typing.Any) -> PT:
+    def __new__(cls: type[PT], *args: str | os.PathLike, conn_id: str | None = None) -> PT:
```

Comment:
using **kwargs here allows us to pass these as (storage) options to the underlying filesystem implementation. This makes us compatible with settings not being set by a `Connection`

Reply:
Ah! I was wondering why UPath takes these. Apparently additional kwargs are used in `storage_options`. We don’t currently provide a way to pass those kwargs anywhere though, so the only way the user can use them is if they create a filesystem manually and call `attach` with it. While that’s a totally valid use case, I’m not sure we need to expose it yet, since the method is not discoverable at all at the moment.

Reply:
To illustrate:

```python
attach(protocol="my", fs=MyFS(foo="xxx"))
base = ObjectStoragePath("my://location/")  # Don't need kwargs anyway...?
```

but

```python
# I can't do this either since my:// is not yet a valid protocol...?
base = ObjectStoragePath("my://location/", foo="xxx")
```

Reply:
Well, I mentioned in the Airflow PR that I wanted to do a follow up pr that passes the kwargs after discussing with @kaxil and I do think it is convenient that you could pass let's say caching options to s3/gcs etc (so not a manually attached fs)

Reply:
How would the user code look like to do that?

Reply:
```python
base = ObjectStoragePath("s3://warehouse", cache="xxx", bla="xxx", anon=True)
```

It might require changing the key for the store cache as otherwise you cannot re-init the fs.

Reply:
I assume these will be forward to `get_fs` and then into the instantiated filesystem, and they would override flags set on Connection (and thus ObjectStore). Is this correct?

This also makes me think maybe we should remove the ability to pass in an expliclt Filesystem object into `attach` since this covers the use case. It would arguably be better security-wise as well since new Filesystem type must then be registered though a plugin or provider.

Reply:
> I assume these will be forward to `get_fs` and then into the instantiated filesystem, and they would override flags set on Connection (and thus ObjectStore). Is this correct?

Yes

> 
> This also makes me think maybe we should remove the ability to pass in an expliclt Filesystem object into `attach` since this covers the use case. It would arguably be better security-wise as well since new Filesystem type must then be registered though a plugin or provider.

This might indeed be a good idea. Especially, since we can serialize the extra options but we cannot (or don't currently) do that for the explicit attachments.